src/interface/TWIoTeXStaking.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include <TrustWalletCore/TWIoTeXStaking.h>
+#include <TrustWalletCore/TWEthereumAbiEncoder.h>
+#include <TrustWalletCore/TWEthereumAbiFunction.h>
+
+#include "Data.h"
+#include "uint256.h"
+
+using namespace TW;
+
+TWData *_Nonnull TWIoTeXStakingStake(TWData *_Nonnull canName, uint64_t stakeDuration, bool nonDecay, TWData *_Nullable data) {
+    TWEthereumAbiFunction* func = TWEthereumAbiEncoderBuildFunction(TWStringCreateWithUTF8Bytes("createPygg"));
```

Comment:
I think you can call `src/Ethereum/ABI/Function` directly, cpp class might be easier to use

Reply:
i did look into src/Ethereum/ABI, but after checking it seems using TWEthereumAbiEncoderxxx is more neat and gives better readability, that is why the current code

for example, TWEthereumAbiFunctionAddParamBytesFix() would translate to 

Data data = TW::data(TWDataBytes(val), TWDataSize(val));
    auto param = std::make_shared<ParamByteArrayFix>(count, data);
    auto idx = function.addParam(param, isOutput);

which is more code lines and worse readability

Reply:
ok, make sure you call necessary delete/clean functions then

Reply:
yes, called TWEthereumAbiEncoderDeleteFunction() before exit